src/middlewared/middlewared/plugins/init_shutdown_script.py

```diff
@@ -156,72 +143,71 @@ async def validate(self, data, schema_name):
 
         verrors.check()
 
+    @private
+    def write_script_text(self, script_text):
+        with tempfile.NamedTemporaryFile(delete=False, mode='w') as f:
+            f.write(script_text)
+            os.fchmod(f.fileno(), stat.S_IRWXU)
+            return f.name
+
+    @private
+    def script_already_exists(self, script):
+        return os.path.exists(script) and os.access(script, os.X_OK)
+
+    @private
+    def rem_tmp_script(self, script):
+        try:
+            os.unlink(script)
+        except FileNotFoundError:
+            pass
+        except Exception:
+            self.logger.debug('Unexpected failure removing %r', script, exc_info=True)
```

Comment:
No need to log the entire traceback here

Reply:
Why?

Reply:
Because we are not interested in who called `rem_tmp_script` (we only have one caller), we are only interested in file path and I/O error.